extensions/amp-analytics/0.1/visibility-model.js

```diff
@@ -79,7 +90,7 @@ export class VisibilityModel {
     this.ready_ = true;
 
     /** @private {boolean} */
-    this.reportReady_ = true;
+    this.reportReady_ = !this.accumulateCounters_;
```

Comment:
This fixes the race condition where the ping is sent right away without waiting on the `reportReadyPromise` because the thread is yielded before `setReportReady` is called. So we'll end up sending a ping if `update` is called too early.

Added a comment clarifying. I'd be happy to cleanup the code in a follow-up PR. What if  `reportReady_` is unconditionally false and the visibility manager calls `setReportReady(true)` or `setReportReady(Promise.resolve())` when `reportWhen` is missing?

Reply:
I agree. Let's address this in following PR, to always have visibility manager setReportReady. Also this could avoid having to check `spec['waitFor']` again in visibility model.